model/adjuster/sort_references.go

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) 2019 The Jaeger Authors.
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package adjuster
+
+import (
+	"sort"
+
+	"github.com/jaegertracing/jaeger/model"
+)
+
+// SortReferences returns an Adjuster that puts CHILD_OF references first.
+// This is necessary to match jaeger-ui expectations:
+// * https://github.com/jaegertracing/jaeger-ui/issues/966
+func SortReferences() Adjuster {
+	return Func(func(trace *model.Trace) (*model.Trace, error) {
+		for _, span := range trace.Spans {
+			sort.SliceStable(span.References, func(i, j int) bool {
+				iType := span.References[i].GetRefType()
+				jType := span.References[j].GetRefType()
+				return iType == model.SpanRefType_CHILD_OF && jType == model.SpanRefType_FOLLOWS_FROM
```

Comment:
I don't think this is a sufficient condition. The logic to pick the correct parent should be:
  * first ref with refType=CHILD_OF and the same trace ID
  * otherwise, first ref that has the same trace ID
  * otherwise no change

Also, I would avoid the actual sorting, and only do the swap into the first position.

Reply:
Makes sense, updated.